tests/unit/gh_pr_upsert/git_test.py

```diff
@@ -0,0 +1,255 @@
+from subprocess import CalledProcessError
+from unittest.mock import sentinel
+
+import pytest
+
+from gh_pr_upsert import git
+
+
+class TestAuthenticatedUsername:
+    def test_it(self, run):
+        run.return_value = {"login": sentinel.username}
+
+        username = git.authenticated_username()
+
+        run.assert_called_once_with(["gh", "api", "/user"], json=True)
+        assert username == sentinel.username
+
+
+class TestPRDiff:
+    def test_it(self, run):
+        diff = git.pr_diff()
+
+        run.assert_called_once_with(["gh", "pr", "diff"])
+        assert diff == run.return_value
+
+
+class TestPR:
+    def test_it_returns_the_open_pr(self, run):
+        run.return_value = {"state": "OPEN"}
+
+        pr = git.pr()
+
+        run.assert_called_once_with(
+            ["gh", "pr", "view", "--json", "state,commits"], json=True
+        )
+        assert pr == run.return_value
+
+    def test_it_returns_None_if_theres_no_open_pr(self, run):
+        run.return_value = {"state": "CLOSED"}
+
+        assert git.pr() is None
+
+    def test_it_returns_None_if_the_command_fails(self, run):
+        run.side_effect = CalledProcessError(
+            returncode=sentinel.returncode, cmd=sentinel.cmd
+        )
+
+        assert git.pr() is None
+
+    @pytest.fixture(autouse=True)
+    def cache_clear(self):
+        git.pr.cache_clear()
```

Comment:
Did you know that when you use `@lru_cache` you need to clear the cache between tests? Otherwise only the first test actually calls the function.